src/Enzo/enzo_EnzoMethodGrackle.cpp

```diff
@@ -235,7 +235,38 @@ void EnzoMethodGrackle::setup_grackle_units (double current_time,
     enzo_float cosmo_dt = 0.0;
 
     EnzoPhysicsCosmology * cosmology = enzo::cosmology();
+    //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+    //put into cosmology units
+    //
+
+    // Conversions are done explicitly here because this function is called in the constructor, which can lead to issues
+    // when calling virtual functions (e.g. enzo_units->length, density, etc.)
+    // Formulae are copied from src/Enzo/enzo_EnzoPhysicsCosmology.hpp
+
+    if (current_time == 0) {
+       current_time = cosmology->time_from_redshift(enzo::config()->physics_cosmology_initial_redshift);
+    }
+
+  double current_redshift = cosmology->redshift_from_time(current_time);
```

Comment:
Minor tweak: fix the indentation.
```suggestion
    double current_redshift = cosmology->redshift_from_time(current_time);
```